lib/fluent/plugin/out_split.rb

```diff
@@ -44,9 +50,18 @@ def split_message(message)
       data = {}
       message.split(@sep_regex).each do |e|
         matched = @format_regex.match(e) or next
-        data.store(matched['key'], matched['value'])
+        @store_fun.call(data, matched['key'], matched['value'])
```

Comment:
ensure prefix has string (at least empty string), then always add prefix to each key instead of calling method.